tests/queries/0_stateless/00873_t64_codec_date.sql

```diff
@@ -0,0 +1,26 @@
+DROP TABLE IF EXISTS t64;
+
+CREATE TABLE t64
+(
+    date16 Date,
+    t_date16 Date Codec(T64, ZSTD),
+    date_32 Date32,
+    t_date32 Date32 Codec(T64, ZSTD)
+) ENGINE MergeTree() ORDER BY tuple();
+
+INSERT INTO t64 values ('1970-01-01', '1970-01-01', '1970-01-01', '1970-01-01');
+INSERT INTO t64 values ('2149-06-06', '2149-06-06', '2149-06-06', '2149-06-06');
+INSERT INTO t64 values ('2149-06-08', '2149-06-08', '2149-06-08', '2149-06-08');
+INSERT INTO t64 values ('1950-01-01', '1950-01-01', '1950-01-01', '1950-01-01');
+
+SELECT * FROM t64 ORDER BY date16;
+
+SELECT * FROM t64 WHERE date16 != t_date16;
+SELECT * FROM t64 WHERE date_32 != t_date32;
+
+OPTIMIZE TABLE t64 FINAL;
+
+SELECT * FROM t64 WHERE date16 != t_date16;
```

Comment:
Looking at the report (https://s3.amazonaws.com/clickhouse-test-reports/58738/998c56fc3d3602a1151c7e310863e12666e595e9/stateless_tests_flaky_check__asan_.html) it seems the test is flaky. It's missing order by.

Reply:
the test case is flaky because we ordered by date16, which has duplicate values on different rows. Changed to `order by date_32`, now it should be stable.